Source/PropertyTools.Wpf/DataGrid/ControlFactory/DataGridControlFactory.cs

```diff
@@ -55,6 +70,23 @@ public virtual FrameworkElement CreateDisplayControl(PropertyDefinition property
         /// </returns>
         public virtual FrameworkElement CreateEditControl(PropertyDefinition propertyDefinition, string bindingPath)
         {
+            if (propertyDefinition is TemplateColumnDefinition)

```

Comment:
I think it is time to refactor so the `CreateEditControl` method is a virtual method on the `PropertyDefinition`. Make it open for extension!